Caronae/Extras/CaronaeImageViewer.swift

```diff
@@ -0,0 +1,58 @@
+import SKPhotoBrowser
+import SDWebImage
+
+class CaronaeImageViewer {
+    static let instance = CaronaeImageViewer()
+
+    private init() {
+        SKCache.sharedCache.imageCache = CustomImageCache()
+        SKPhotoBrowserOptions.displayAction = false
+        SKPhotoBrowserOptions.bounceAnimation = true
+        SKPhotoBrowserOptions.displayStatusbar = true
+    }
+    
+    func present(pictureURL: String, animatedFrom view: UIImageView) {
+        let photo = SKPhoto.photoWithImageURL(pictureURL)
+        photo.shouldCachePhotoURLImage = true
+        
+        if let topViewController = UIApplication.shared.topViewController() {
+            let browser = CustomSKPhotoBrowser(originImage: view.image ?? UIImage(), photos: [photo], animatedFromView: view)
+            topViewController.present(browser, animated: true)
+        }
+    }
+}
+
+
+// MARK: Custom classes
+
+fileprivate class CustomSKPhotoBrowser: SKPhotoBrowser {
+    
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return .lightContent
+    }
+}
+
+
+fileprivate class CustomImageCache: SKImageCacheable {
```

Comment:
Eu acho que a gente não precisa lidar com cache aqui. 

Essa imagem é a mesma que vem da miniatura, da outra biblioteca. E aquela lib já tem seu próprio controle de cache. Acho que só faria sentido se a gente tivesse carregando uma imagem diferente aqui (tipo se a miniatura fosse em baixa resolução e aí carrega em alta definição só quando clica), mas não temos isso. Então se isso for verdade acho que a gente acaba criando um segundo cache da mesma imagem.

Tipo, o único caso que eu consigo ver isso sendo útil é caso o usuário clique pra ver a foto em tela cheia quando a miniatura não carregou ainda (no caso da `view.image` não ter sido carregada ainda mesmo). Mas acho isso bem improvável de acontecer. 

Mas sei lá, não é nada demais também hahaha

Reply:
Então, eu tava pensando justamente nesse caso do usuário clicar na foto antes de carregar, não sei se com a internet ruim corre o risco de acontecer.

Mas a sacada é que essa classe tá configurada pra utilizar o cache da SDWebImage. Então a gente passa a url da imagem, se já tiver no cache da SDWebImage ele pega de lá, se não ele carrega pelo link e adiciona na SDWebImage.

Aí acho que dessa forma não tá ficando um segundo cache da mesma imagem. Da pra fazer o teste, por exemplo, colocando no modo avião depois da miniatura ter carregado e tentar abrir a foto. Mas a gente pode repensar ou então deixar passando só a image mesmo. O parâmetro de view que tá recebendo por enquanto só serve pra fazer a animação de zoom quando clica na foto mesmo.

Reply:
Entendi. Bom, então acho que é isso mesmo! Boa sacada em usar o cache de uma lib pra outra :D 

Eu não cheguei a testar essa branch, mas em relação ao código por mim tá show, podem mergear!